### PR TITLE
Set environment variable IS_CI

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/377271824

Set IS_CI environment variable as discussed in go/cobaltgnpy

### CODE REVIEW

File: cobalt/docker/linux/Dockerfile [hunk 1]
```diff
@@ -34,3 +34,4 @@ ENV SCCACHE_GCS_BUCKET=githubactions-chrome-sccache
 ENV SCCACHE_GCS_RW_MODE=READ_WRITE
 ENV SCCACHE_GCS_KEY_PATH=/root/key.json
 ENV SCCACHE_GCS_SERVICE_ACCOUNT=github-actions-bucket-access@cobalt-demo-330821.iam.gserviceaccount.com
+ENV IS_CI=1
```
Comment (kaidokert) [inline cobalt/docker/linux/Dockerfile:37]: Technically this line should belong in `.github/` yaml code somewhere, that passes it to container via `-e` environment variable flag. "I am in a container" is not synonymous with "i am running in CI"

But given `sccache` config lives here for no, we can leave it - perhaps leave a todo to factor both out later.

**Review by kaidokert — APPROVED** [review]

File: cobalt/build/gn.py [hunk 2]
```diff
@@ -84,10 +84,10 @@ def write_build_args(build_args_path, original_lines, dict_settings,
         f'The following args cannot be set in configs: {controlled_args}')
   gen_comment = '# Set by gn.py'
   with open(build_args_path, 'w', encoding='utf-8') as f:
-    if os.getenv('USE_RBE', default='0') == '0':
-      f.write(f'cc_wrapper = "{CC_WRAPPER}" {gen_comment}\n')
-    else:
+    if os.getenv('IS_CI', default='0') == '0':
```
Comment (andrewsavage1) [inline cobalt/build/gn.py:87]: While this isn't really a negative conditional, it kind of feels like one; I have a slight preference for `if os.getenv('IS_CI', default='0') == '1'` (which to me does what it reads like at a glance, i.e. `if IS_CI`)

**Review by andrewsavage1 — APPROVED** [review]

File: cobalt/docker/linux/Dockerfile (hunk 1 above)
Comment (niranjanyardi) [reply cobalt/docker/linux/Dockerfile:38]: Done, added TODO bug

File: cobalt/build/gn.py (hunk 2 above)
Comment (niranjanyardi) [reply cobalt/build/gn.py]: changed conditional as requested.

**Pull request merged**